ecosystem/python/sdk/examples/rotate-key.py

```diff
@@ -0,0 +1,109 @@
+    from_account: Account,
+    private_key: PrivateKey
+) -> TransactionPayload:
+    to_account = Account.load_key(private_key.hex())
+    rotation_proof_challenge = RotationProofChallenge(
+        sequence_number=await rest_client.account_sequence_number(from_account.address()),
+        originator=from_account.address(),
+        current_auth_key=AccountAddress.from_hex(from_account.auth_key()),
+        new_public_key=to_account.public_key().key.encode(),
+    )
+
+    serializer = Serializer()
+    rotation_proof_challenge.serialize(serializer)
+    rotation_proof_challenge_bcs = serializer.output()
+
+    proof_signed_by_from = from_account.sign(rotation_proof_challenge_bcs).data()
+    proof_signed_by_to = to_account.sign(rotation_proof_challenge_bcs).data()
+
+    from_scheme = Authenticator.ED25519
+    from_public_key_bytes = from_account.public_key().key.encode()
+    to_scheme = Authenticator.ED25519
+    to_public_key_bytes = to_account.public_key().key.encode()
+
+    entry_function = EntryFunction.natural(
+        module="0x1::account",
+        function="rotate_authentication_key",
+        ty_args=[],
+        args=[
+            TransactionArgument(from_scheme, Serializer.u8),
+            TransactionArgument(from_public_key_bytes, Serializer.to_bytes),
+            TransactionArgument(to_scheme, Serializer.u8),
+            TransactionArgument(to_public_key_bytes, Serializer.to_bytes),
+            TransactionArgument(proof_signed_by_from, Serializer.to_bytes),
+            TransactionArgument(proof_signed_by_to, Serializer.to_bytes),
+        ],
+    )
+
+    return TransactionPayload(entry_function)
+
+
+async def main():
+    rest_client = RestClient(NODE_URL)
+    faucet_client = FaucetClient(FAUCET_URL, rest_client)
+    alice = Account.generate()
+    bob = Account.generate()
+
+    await faucet_client.fund_account(alice.address(), 10_000_000)
+    await faucet_client.fund_account(bob.address(), 10_000_000)
+    print('\n' + 'Account'.ljust(WIDTH, " ") + 'Address'.ljust(WIDTH, " ") + 'Auth Key'.ljust(WIDTH, " ") + 'Private Key'.ljust(WIDTH, " ") + 'Public Key'.ljust(WIDTH, " "))
+    print('-------------------------------------------------------------------------------------------')
+    print('Alice'.ljust(WIDTH, " ") + format_account_info(alice))
+    print('Bob'.ljust(WIDTH, " ") + format_account_info(bob))
+
+    print("\n...rotating...\n")
+
+    # :!:>rotate_key
+    payload = await rotate_auth_key_ed_25519_payload(rest_client, alice, bob.private_key)
+
+    signed_transaction = await rest_client.create_bcs_signed_transaction(
+        alice, payload
+    ) # <:!:rotate_key
+
+    tx_hash = await rest_client.submit_bcs_transaction(signed_transaction)
+    await rest_client.wait_for_transaction(tx_hash)
+
+    new_account_info = await rest_client.account(alice.address())
+    assert AccountAddress.from_hex(new_account_info['authentication_key']) == bob.address(), "Authentication key doesn't match Bob's address"
```

Comment:
This is a little confusing here, maybe put that you're rotating Alice to Bob, and that the new authentication key should be the same as the authentication key of Bob's rather than the address.